spec/features/search/chat_channel_membership_spec.rb

```diff
@@ -0,0 +1,94 @@
+require "rails_helper"
+
+RSpec.describe Search::ChatChannelMembership, type: :feature, elasticsearch: true do
```

Comment:
I really wanted solid end-to-end search specs bc in the past I have found these are BY FAR the most useful at catching bugs when adding new code. Yes, they are slower but trust me, we want these. I choose to make a new feature folder for them. If someone has another suggested approach let me know.

Reply:
I'm a little bit confused about this choice, in the sense that for RSpec feature specs are basically system tests designed to excercise the app from the outside: https://relishapp.com/rspec/rspec-rails/docs/feature-specs/feature-spec

but these are normal services specs.

What about just moving this in `spec/services/search`? You can either add them to `chat_channel_membership_spec.rb` in a `describe .search_documents` section or add a separate file like `chat_channel_membership_search_documents.rb` or `spec/services/search/chat_channel_membership/search_documents.rb`, your call :)

Reply:
I thought of them as feature specs bc they aren't just testing the service's ability to talk to Elasticsearch they are also testing the query builders ability to form a valid query and then that the query returns valid results. It felt a little weird bc if we add another filter key to query builder, this is where we would want to add another spec which is what led me to consider them "feature" specs as opposed to service specs but I am game for consolidating!

Reply:
Got it! +1 for consolidating. Most service specs test features anyway. I think it's just a matter of naming, as what we call "feature spec" is not what RSpec calls "feature spec" :D